dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/AppConfigManager.java

```diff
@@ -178,28 +162,41 @@ public void setAttenuationFactorMedium(float factor) {
 		sharedPrefs.edit().putFloat(PREF_ATTENUATION_FACTOR_MEDIUM, factor).apply();
 	}
 
-	public HashMap<DayDate, Long> getLastLoadedTimes() {
-		return convertToDateMap(Json.fromJson(sharedPrefs.getString(PREF_LAST_LOADED_TIMES, "{}"), StringLongMap.class));
+	public int getNumberOfDaysToConsiderForExposure() {
+		return sharedPrefs.getInt(PREF_NUMBERR_OF_DAYS_TO_CONSIDER_FOR_EXPOSURE, DEFAULT_NUMBER_OF_DAYS_TO_CONSIDER_FOR_EXPOSURE);
 	}
 
-	public HashMap<DayDate, Long> getLastSyncCallTimes() {
-		return convertToDateMap(Json.fromJson(sharedPrefs.getString(PREF_LAST_SYNC_CALL_TIMES, "{}"), StringLongMap.class));
+	public void setNumberOfDaysToConsiderForExposure(int days) {
+		sharedPrefs.edit().putInt(PREF_NUMBERR_OF_DAYS_TO_CONSIDER_FOR_EXPOSURE, days).apply();
 	}
 
-	public HashMap<DayDate, Long> getLastSuccessfulSyncTimes() {
-		return convertToDateMap(Json.fromJson(sharedPrefs.getString(PREF_LAST_SUCCESSFUL_SYNC_TIMES, "{}"), StringLongMap.class));
+	public int getNumberOfDaysToKeepExposedDays() {
+		return sharedPrefs.getInt(PREF_NUMBER_OF_DAYS_TO_KEEP_EXPOSED_DAYS, DEFAULT_NUMBER_OF_DAYS_TO_KEEP_EXPOSED_DAYS);
 	}
 
-	public void setLastLoadedTimes(HashMap<DayDate, Long> lastLoadedTimes) {
-		sharedPrefs.edit().putString(PREF_LAST_LOADED_TIMES, Json.toJson(convertFromDateMap(lastLoadedTimes))).apply();
+	public void setNumberOfDaysToKeepExposedDays(int days) {
+		sharedPrefs.edit().putInt(PREF_NUMBER_OF_DAYS_TO_KEEP_EXPOSED_DAYS, days).apply();
 	}
 
-	public void setLastSyncCallTimes(HashMap<DayDate, Long> lastExposureClientCalls) {
-		sharedPrefs.edit().putString(PREF_LAST_SYNC_CALL_TIMES, Json.toJson(convertFromDateMap(lastExposureClientCalls))).apply();
+	public long getLastSynCallTime() {
+		return sharedPrefs.getLong(PREF_LAST_SYNC_CALL_TIME, 0);
 	}
 
-	public void setLastSuccessfulSyncTimes(HashMap<DayDate, Long> lastSuccessfulSyncTimes) {
-		sharedPrefs.edit().putString(PREF_LAST_SUCCESSFUL_SYNC_TIMES, Json.toJson(convertFromDateMap(lastSuccessfulSyncTimes))).apply();
+	public void setLastSyncCallTime(long time) {
+		sharedPrefs.edit().putLong(PREF_LAST_SYNC_CALL_TIME, time).apply();
+	}
+
+	public Long getLastKeyBundleTag() {
```

Comment:
I'd prefer switching to a string for the key bundle tag to have all options open in the backend to change this to something else.

Reply:
Agreed! Has been changed to use a String value!